test/test.js

```diff
@@ -18,6 +18,20 @@ async function run() {
       process.exit(0)
     }
   }, 40 * 1000)
+
+  console.log('try connecting to server that does not exist')
+  let badClient = new WebTransport('https://127.0.0.1:49823/echo', {
+    serverCertificateHashes: [{ algorithm: 'sha-256', value: Buffer.from('a589bf4f98a0158aa890328d5d3f519b9e2a5b1e61b09eb10b7a9be0e79bf148', 'hex') }]
+  })
+  await badClient.ready
+    .then(() => {
+      console.error('Successfully connected to a non-running server?!')
+      process.exit(1)
+    })
+    .catch(() => {
+      console.log('Did not connect to non-running server')
```

Comment:
Would be nice to assert on the thrown error here - `.code` property or `.message` but no error is throw?

Reply:
There is currently no mechanism in `Http3Client`.
Probably a time out in `handleConnecting` together with code to tear down the connection in an error case.
But agreed, should be done. However currently I want first to write the code, that needs the time out.

Reply:
But you can program a timeout yourself on the js side and close after time out.
Something like setTimeout(()=>{if (notconnected) session.close()},10000), so I do not see urgency.

Reply:
After being more awake. I think the main problem is that no one is responding. Since the underlying UDP protocol is connectionless, the only way to detect that no one is responding is via timeout in the handshakes in the client code.
The question is what should be the default timeout?

Reply:
From experimentation it's 4s in Chrome.  I can't see anything in the spec about overriding default timeouts though, which raises an eyebrow slightly.

Reply:
Since it is not in the specs, we would deviate, if we allow it to change.
We can implement it easily on the c++ side, (but my development dir is currently tied to the macOS arm build...., so not now), but then it is fixed, or we allow to pass a different value at construction time extending the spec.